client/ayon_resolve/plugins/publish/collect_editorial_package.py

```diff
@@ -1,88 +1,51 @@
-import json
-
 import pyblish.api
 
 import ayon_api
-from ayon_api import get_task_by_id
 
-from ayon_resolve.api import lib
+from ayon_resolve.api import lib, constants
 
 
-class EditorialPackageInstances(pyblish.api.ContextPlugin):
+class EditorialPackageInstances(pyblish.api.InstancePlugin):
     """Collect all Track items selection."""
 
     order = pyblish.api.CollectorOrder - 0.49
     label = "Collect Editorial Package Instances"
     families = ["editorial_pkg"]
 
-    def process(self, context):
-        project_name = context.data["projectName"]
+    def process(self, instance):
+        project_name = instance.context.data["projectName"]
         self.log.info(f"project: {project_name}")
 
-        for media_pool_item in lib.iter_all_media_pool_clips():
-
-            data = media_pool_item.GetMetadata(lib.pype_tag_name)
-            if not data:
-                continue
-
-            try:
-                data = json.loads(data)
-            except json.JSONDecodeError:
-                self.log.warning(
-                    f"Failed to parse json data from media pool item: "
-                    f"{media_pool_item.GetName()}"
-                )
-                continue
-
-            # exclude all which are not productType editorial_pkg
-            if (
-                data.get("publish")
-                and data["publish"].get("productType") != "editorial_pkg"
-            ):
-                continue
-
-            instance = context.create_instance(name=media_pool_item.GetName())
+        media_pool_item = instance.data["transientData"]["timeline_pool_item"]
 
-            publish_data = data["publish"]
+        # get version from publish data and rise it one up
```

Comment:
Since this logic existed before this PR and this matches it - let's keep it in. It would be nice though @robin-ynput if here you could elaborate on what this does and why (like a small elaboration on it).

And then separate from this PR let's create an issue to track that we will want to change this logic. It's a very poor-man's solution to what should not be so specific to this particular instance/product.

If we want _that type of versioning_ we should document that clearly, define the strategy and potentially make that "just one of the ways you can version products" in AYON. Now it just seems way too hardcoded for just this resolve use case (which I still think isn't the correct use case, but that's up for debate in a separate issue.)

Reply:
As per our discussion, relevant comment has been added.